test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1

```diff
@@ -5,219 +5,488 @@
+        $result.count | Should Be $testCase.Count
+        $result | Should BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
+        $result[0].ToString() | Should Match $testCase.ExpectedResult
     }
+}
 
-    # This test is to validate to input given through pipeline. The input being piped from results of Get-hildItem
-    It "ValidateFileInfoPipelineInput" {
+function RunContinuesToProcessCase($testCase)
+{
+    It "$($testCase.Name)" {
 
-        $result = Get-ChildItem $inputFile1 | Format-Hex
-        $result | Should Not BeNullOrEmpty
-        ,$result | Should BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
-        $actualResult = $result.ToString()
-        ($actualResult -match $inputText1) | Should Be $true
-    }
+        $output = $null
+        $errorThrown = $null
 
-    # This test is to validate Encoding formats functionality of Format-Hex cmdlet.
-    It "ValidateEncodingFormats" {
+        if ($testCase.PathCase)
+        {
+            $output = Format-Hex -Path $testCase.InvalidPath, $inputFile1 -ErrorVariable errorThrown -ErrorAction SilentlyContinue
+        }
+        else # LiteralPath
+        {
+            $output = Format-Hex -LiteralPath $testCase.InvalidPath, $inputFile1 -ErrorVariable errorThrown -ErrorAction SilentlyContinue
+        }
 
-        $result =  Format-Hex -InputObject $inputText1 -Encoding ASCII
-        $result | Should Not BeNullOrEmpty
-        ,$result | Should BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
-        $actualResult = $result.ToString()
-        ($actualResult -match $inputText1) | Should Be $true
+        $errorThrown | Should Not Be $null
+        $errorThrown.FullyQualifiedErrorId | Should Match $testCase.ExpectedFullyQualifiedErrorId
+
+        $output | Should Not Be $null
+        $output[0].ToString() | Should Match $inputText1
     }
+}
 
-    # This test is to validate that integers can be piped to the format-hex
-    It "ValidateIntegerInput" {
+function RunExpectedErrorTestCase($testCase)
+{
+    It "$($testCase.Name)" {
 
-        $result = 1,2,3,4 | Format-Hex
-        $result | Should Not BeNullOrEmpty
-        ,$result | Should BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
-        $actualResult = $result.ToString()
-        # whitespace sensitive
-        $actualResult | should be "00000000   01 02 03 04                                      ....            "
+        try
+        {
+            if ($testCase.PathParameterErrorCase)
+            {
+                $result = Format-Hex -Path $testCase.Path -ErrorAction Stop
+            }
+            if ($testCase.InputObjectErrorCase)
+            {
+                $result = Format-Hex -InputObject $testCase.InputObject -ErrorAction Stop
+            }
+        }
+        catch
+        {
+            $thrownError = $_
+        }
+
+        $thrownError | Should Not Be $null
+        $thrownError.FullyQualifiedErrorId | Should Match $testCase.ExpectedFullyQualifiedErrorId
```

Comment:
I've tried using this but get this error:
The term 'ShouldBeErrorId' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.

So for now I'll keep Should Match